infra/staging/main.tf

```diff
@@ -152,6 +152,99 @@ resource "google_storage_bucket_iam_member" "frontend_bucket_viewer" {
   member = "allUsers"
 }
 
+#######################################
+## Load balancer for frontend bucket ##
+#######################################
+
+// Following the guide here: https://cloud.google.com/cdn/docs/setting-up-cdn-with-bucket
+
+// Enable Compute Engine API to setup an IP address
+resource "google_project_service" "compute_engine_api" {
+  project = data.google_project.project.project_id
+  service = "compute.googleapis.com"
+}
+
+// Setup IP address for load balancer
+resource "google_compute_global_address" "call_home_frontend_ip" {
+  project = data.google_project.project.project_id
+  name = "call-home-frontend-ip"
+}
+
+// Backend bucket with CDN policy with default ttl settings
+resource "google_compute_backend_bucket" "call_home_frontend_lb" {
+  name = "call-home-frontend-lb"
+  description = "Load balancer for the frontend GCS bucket"
+  project = data.google_project.project.project_id
+  bucket_name = google_storage_bucket.call_home_frontend.name
+  enable_cdn = true
+
+  cdn_policy {
+    cache_mode = "CACHE_ALL_STATIC"
+    client_ttl = 3600
+    default_ttl = 3600
+    max_ttl = 86400
+    negative_caching = true
+    serve_while_stale = 86400
+  }
+}
+
+// URL map for the backend bucket
+resource "google_compute_url_map" "call_home_lb" {
+  name = "call-home-lb"
+  project = data.google_project.project.project_id
+  default_service = google_compute_backend_bucket.call_home_frontend_lb.id
+
+  depends_on = [
+    google_compute_backend_bucket.call_home_frontend_lb
+  ]
+}
+
+// Setup SSL certificate to enable HTTPS for load balancer
+resource "google_compute_managed_ssl_certificate" "call_home_frontend_lb_ssl" {
+  name = "call-home-frontend-lb-ssl"
+  project = data.google_project.project.project_id
+
+  managed {
+    domains = ["app2-staging.callhome.sg"]
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+// HTTP proxy
+resource "google_compute_target_https_proxy" "call_home_frontend_lb_proxy" {
+  name = "call-home-frontend-lb-proxy"
+  project = data.google_project.project.project_id
+  url_map = google_compute_url_map.call_home_lb.id
+
+  ssl_certificates = [
+    google_compute_managed_ssl_certificate.call_home_frontend_lb_ssl.name
+  ]
+
+  depends_on = [
+    google_compute_managed_ssl_certificate.call_home_frontend_lb_ssl,
+    google_compute_url_map.call_home_lb
+  ]
+}
+
+// Forwarding rule
+resource "google_compute_global_forwarding_rule" "call_home_frontend_lb_forwarding" {
+  name = "call-home-frontend-lb-forwarding"
+  project = data.google_project.project.project_id
+  ip_protocol = "TCP"
+  load_balancing_scheme = "EXTERNAL"
+  port_range = "443"
```

Comment:
Ah, as of this version I see that we no longer have a rule for `http://`. As a result, http://app2-staging.callhome.sg/ no longer works.

What we probably want is a typical HTTP -> HTTPS redirect. It doesn't look too hard to set up, hopefully.
https://cloud.google.com/load-balancing/docs/https/setting-up-http-https-redirect
https://stackoverflow.com/questions/64375885/google-cloud-forwarding-rule-http-https-using-terraform